sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ViewCatalog.java

```diff
@@ -140,6 +140,92 @@ View createView(
       String[] columnComments,
       Map<String, String> properties) throws ViewAlreadyExistsException, NoSuchNamespaceException;
 
+  /**
+   * Replace a view in the catalog.
+   * <p>
+   * The default implementation has a race condition.
+   * Catalogs are encouraged to implement this operation atomically.
+   *
+   * @param ident a view identifier
+   * @param sql the SQL text that defines the view
+   * @param currentCatalog the current catalog
+   * @param currentNamespace the current namespace
+   * @param schema the view query output schema
+   * @param queryColumnNames the query column names
+   * @param columnAliases the column aliases
+   * @param columnComments the column comments
+   * @param properties the view properties
+   * @throws NoSuchViewException If the view doesn't exist or is a table
+   * @throws NoSuchNamespaceException If the identifier namespace does not exist (optional)
+   */
+  default void replaceView(
+          Identifier ident,
+          String sql,
+          String currentCatalog,
+          String[] currentNamespace,
+          StructType schema,
+          String[] queryColumnNames,
+          String[] columnAliases,
+          String[] columnComments,
+          Map<String, String> properties) throws NoSuchViewException, NoSuchNamespaceException {
+    if (viewExists(ident)) {
+      dropView(ident);
+      try {
+        createView(ident, sql, currentCatalog, currentNamespace, schema,
+                queryColumnNames, columnAliases, columnComments, properties);
+      }
+      catch (ViewAlreadyExistsException e) {
+        throw new RuntimeException("Race condition when dropping and creating view", e);
+      }
+    } else {
+      throw new NoSuchViewException(ident);
+    }
+  }
+
+  /**
+   * Create or replace a view in the catalog.
+   * <p>
+   * The default implementation has race conditions.
+   * Catalogs are encouraged to implement this operation atomically.
+   *
+   * @param ident a view identifier
+   * @param sql the SQL text that defines the view
+   * @param currentCatalog the current catalog
+   * @param currentNamespace the current namespace
+   * @param schema the view query output schema
+   * @param queryColumnNames the query column names
+   * @param columnAliases the column aliases
+   * @param columnComments the column comments
+   * @param properties the view properties
+   * @throws NoSuchNamespaceException If the identifier namespace does not exist (optional)
+   */
+  default void createOrReplaceView(
```

Comment:
3 APIs might be too much. How about we follow the create/replace table logical plan? We have a `CreateTable` with a bool field `ignoreIfExists`, and a `ReplaceTable` with a bool field `orCreate`.

So we can have just two methods here
```
def createView(... , ignoreIfExists: Boolean)

def replaceView(..., orCreate: Boolean)
```

Reply:
thanks for the feedback @cloud-fan. I'll follow up with your suggestions and will ping you for a review

Reply:
@cloud-fan I have made the suggested improvements in https://github.com/apache/spark/pull/44330 but I'm not sure about having `ignoreIfExists` on the API when creating a view. 
I checked and the `TableCatalog` also doesn't have such functionality and I wonder if it wouldn't be better to have `ignoreIfExists` be part of the physical plan that creates the view (similar to how it's done in https://github.com/apache/spark/pull/44197/files#diff-d491b3b4f1ac890067f57f4c30cbdecf3585ea91dabf0dced59b31a4e1643462R83)

Reply:
Makes sense. A simpler API is better. The caller side (Spark) can just ignore the error if `ignoreIfExists` is true.